simstore_and_cli/2_running_cli.md

```diff
@@ -0,0 +1,203 @@
+state transition interface sampling, because a trajectory that has visited all
+states is guaranteed to have subtrajectories that meet the requirements of
+every TIS ensemble in MSTIS. This is also true for any two-state system, such
+as the one we are studying here.
+
+To generate a trajectory with the `VisitAllStatesEnsemble`, use the `visit-all`
+subcommand. You can investigate it's options with `openpathsampling visit-all
+--help`: we will need to provide the name of an engine, the names of the
+states, and an output file. In principle, we would need to provide an initial
+frame as well, but OPS can infer the initial conditions since there is only one
+snapshot saved to storage.
+
+```bash
+openpathsampling visit-all setup.db -s condensed -s extended -e hi_temp -o visit-all.db
+```
+
+The resulting trajectory is saved in the file `visit-all.db` with the tag
+`final_conditions`. If we want to copy that trajectory to `setup.db`, we can
+use the `append` subcommand. We will change the tag name it is saved under to
+`long_traj`:
+
+```bash
+openpathsampling append visit-all.db -a setup.db --tag final_conditions --save-tag long_traj
+```
+
+## Filling in the minus interface ensemble
+
+At this point, if we tried to run the final `pathsampling` command (or even the
+`equilibrate` command) we would get an error:
+
+```pytb
+AssertionError: Bad initial conditions.
+Missing ensembles:
+*  [condensed MIS minus]
+```
+
+This is because our TIS network includes a minus ensemble associated with the
+condensed state. Since our `visit-all` simulation started in the extended
+state, it stopped as soon as it found its first frame in the condensed state.
+No subtrajectory of that trajectory will meet the requirements of the minus ensemble.
+
+However, by running dynamics starting in that last frame, we will (very likely)
+obtain a trajectory that satisfies minus ensemble. (Note that there is a small
+chance that the trajectory will spontaneously cross to the other state, which
+would not satisfy the minus ensemble.)
+
+While this could be done by running standard MD for a fixed period of time, and
+checking if the resulting trajectory met the minus ensemble requirements, the
+OPS CLI provides an easier approach: the `md` subcommand, which can run MD for
+either a fixed time or until a given ensemble's criteria are met.
+
+The `md` subcommand will require that we provide the ensemble name, the engine,
+and the initial frame. Since we happen to know that the frame we want is the
+last one that was saved, we can select it with `-1`. For the ensemble, we can
+see the name that was listed as missing in the error, or we can use the command
+`openpathsampling contents setup.db --table ensembles` to see all the named
+ensembles in our setup file. Noting that the ensemble name should be in quotes
+(to preserve the spaces), the command to run the MD is:
+
+```bash
+openpathsampling md setup.db -o minus-traj.db -e engine -f -1 --ensemble "condensed MIS minus"
+```
```

Comment:
Just FYI: this ran into max_length twice before succeeding for me

Reply:
Yeah, I've added a comment about that here. I think what's happening here is that you're getting a recrossing -- the kinetic energy on entering condensed state isn't dissipated quickly enough, and you bounce back to the extended state. This tended to be a serious problem with this system in NVE, but even in NVT it seems to occur on occasion. When I've had max-length errors happen here, I've noticed that it tends to repeat -- this leads me to believe that the initial conditions are bad.

If this continues to be a frequent problem, I might also increase the collision rate in the integrator, which should help.

(The original TIS paper used a 2D version of this system in NVE. The state definitions included a kinetic energy component to avoid these recrossings.)